src/Kralizek.Extensions.Configuration.AWSSecretsManager/Internal/SecretsManagerConfigurationProviderOptions.cs

```diff
@@ -20,5 +20,25 @@ public class SecretsManagerConfigurationProviderOptions
         public Func<IAmazonSecretsManager>? CreateClient { get; set; }
 
         public TimeSpan? PollingInterval { get; set; }
+
+        /// <summary>
+        /// Determines whether or not the options for the configuration provider should be read from configuration.
+        /// </summary>
+        public bool ReadFromConfig { get; set; }
+        
+        /// <summary>
+        /// The name of the configuration section containing the options.
+        /// </summary>
+        public string ConfigSectionName { get; set; } = SecretsManagerConfigurationSection.DefaultConfigSectionName;
+        
+        /// <summary>
+        /// Enables the extension to read options from an IConfigurationSection.
+        /// </summary>
+        /// <param name="configSectionName">The name of the configuration section containing the options.</param>
+        public void ReadFromConfigSection(string configSectionName = SecretsManagerConfigurationSection.DefaultConfigSectionName)
```

Comment:
I'd go for `ReadFromConfiguration` instead.

```suggestion
        public void ReadFromConfiguration(string configSectionName = SecretsManagerConfigurationSection.DefaultConfigSectionName)
```

Another option could be `UseConfiguration`. What do you think?